src/sentry_core.c

```diff
@@ -826,6 +826,19 @@ sentry_remove_fingerprint(void)
     }
 }
 
+void
+sentry_set_trace_id(const char *trace_id)
```

Comment:
`char*` is most probably not the appropriate type here. Since this is supposed to mimic the header-parsing functionality from `sentry_tracing` we could pull that out from there and reuse it here?